conwayste/src/ui/ui_errors.rs

```diff
@@ -15,11 +15,33 @@
  *  You should have received a copy of the GNU General Public License
  *  along with libconway.  If not, see <http://www.gnu.org/licenses/>. */
 
+use crate::ggez::GameError;
+
 custom_error! {pub UIError
     InvalidDimensions {reason: String} = "UIError::InvalidDimensions({reason})",
     WidgetNotFound {reason: String} = "UIError::WidgetNotFound({reason})",
     InvalidAction {reason: String} = "UIError::InvalidAction({reason})",
-    ActionRestricted{reason: String} = "UIError::ActionRestricted({reason})"
+    ActionRestricted{reason: String} = "UIError::ActionRestricted({reason})",
+    WidgetIDCollision{reason: String} = "UIError::WidgetIDCollision({reason})",
+    GgezError{reason: String} = "UIError::GgezError({reason})"
 }
 
 pub type UIResult<T> = Result<T, Box<UIError>>;
```

Comment:
Just a note -- we will probably need to replace this with `CwResult` which would be `Result<T, Box<dyn Error>>`. There is no benefit to having a concrete type in `Box<...>` here.

Not something for this PR though -- it'll be more far reaching changes.

Reply:
Yeah I agree with you now that you've pointed this out. I've created #87 to track this.